### PR TITLE
Zendesk reactor: skip getting zendesk tickets if user has no zendesk account

### PR DESCRIPTION
**Pull request by RashaMou**
@wolfsage I am not sure how to test this locally.

### CODE REVIEW

**Comment by RashaMou** [comment]
@wolfsage this is ready for review

**Comment by wolfsage** [comment]
This should log if we don't find a user, and also log if there's some kind of exception. This way we can see why reports might not fire for a user and not have to guess

**Review by wolfsage — CHANGES_REQUESTED** [review]

File: lib/Synergy/Reactor/Zendesk.pm [hunk 1]
```diff
@@ -253,6 +254,26 @@ async sub ticket_report ($self, $who, $arg = {}) {
...
+  if ($@) {
+    my $error = $@;
+    $Logger->log([
+      "Error while attempting to get Zendesk user for %s: %s",
+      $email,
+      $error
+    ]);
+  };
+
+  unless ($user_ok) {
```
[the remark this answers is not in the document]
Comment (RashaMou) [reply lib/Synergy/Reactor/Zendesk.pm]: yes, it does, thanks

File: lib/Synergy/Reactor/Zendesk.pm [hunk 2]
```diff
@@ -253,11 +254,31 @@ async sub ticket_report ($self, $who, $arg = {}) {
...
+                ]);
+    return;
+  };
+
   my $res;
   my $ok = eval {
     $res = await $self->zendesk_client
                       ->user_api
-                      ->get_by_email_no_fetch($email)
+                      ->$user
```
Comment (wolfsage) [inline lib/Synergy/Reactor/Zendesk.pm:281]: This change makes no sense. Really what you'd do at this point is `$res = $user->scoped_client->make_request_f...` probably. Maybe we pair on this this week and do it live on synergy and see what happens :)

Comment (RashaMou) [reply lib/Synergy/Reactor/Zendesk.pm:281]: Yeah that would be great, there's something I'm not quite getting here!

**Pull request merged**